main.cpp

```diff
@@ -49,77 +49,105 @@ Create a branch named Part1
-    }
+    T* compare(T* a, T* b)
+    {   
+        if (a != nullptr && b != nullptr) 
+        {
+            if( a->value < b->value ) return a;
+            if( a->value > b->value ) return b;
+        }
+        return nullptr; 
+    } 
 };
 
 struct U
 {
-    float <#name1#> { 0 }, <#name2#> { 0 };
-    <#returnType#> <#memberFunction#>(<#type name#>* <#updatedValue#>)      //12
+    float var1 { 0 }, var2 { 0 };
+    
+    U(float a) : var1 (a) { }
+    
+    U();
+    
+    float product( float* inputValue ) 
     {
-        
+        std::cout << "\n(non-static) U's var1 value: " << this->var1 << std::endl;
+        if (inputValue != nullptr) { this->var1 = *inputValue;}
+        std::cout << "\n(non-static) U's var1 updated value: " << this->var1 << std::endl;
+        std::cout << "Reducing magnitude.. \n";
+        while( std::abs (static_cast <long> (this->var2 - this->var1)) > 0.001f )
+        {
+            /*
+             write something that makes the distance between that->name2 and that->name1 get smaller
+             */
+            this->var2 += 0.5f;
+            std::cout << this->var2 << ", ";
+
+        }
+        std::cout << "\n(non-static) U's var2 updated value: " << this->var2 << std::endl;
+        return this->var2 * this->var1;
     }
 };
 
-struct <#structname2#>
+struct StaticTransform
 {
-    static <#returntype#> <#staticFunctionA#>(U* that, <#type name#>* <#updatedValue#> )        //10
+    static float product(U* that, float* inputValue ) 
     {
-        std::cout << "U's <#name1#> value: " << that-><#name1#> << std::endl;
-        that-><#name1#> = <#updatedValue#>;
-        std::cout << "U's <#name1#> updated value: " << that-><#name1#> << std::endl;
-        while( std::abs(that-><#name2#> - that-><#name1#>) > 0.001f )
+        if ( inputValue != nullptr && that != nullptr)
         {
-            /*
-             write something that makes the distance between that-><#name2#> and that-><#name1#> get smaller
-             */
-            that-><#name2#> += ;
+            std::cout << "\nU's var1 value: " << that->var1 << std::endl;
+            that->var1 = *inputValue; 
+            std::cout << "\nU's var1 updated value: " << that->var1 << std::endl;
+            std::cout << "Reducing magnitude.. \n";
+            while( std::abs (static_cast <long> (that->var2 - that->var1)) > 0.001f )
+                {
+                    /*
+                    write something that makes the distance between that->name2 and that->name1 get smaller
+                     */          
+                    that->var2 += 0.5f;
+                    std::cout << that->var2 << ", ";
+                }
+        std::cout << "\nU's var2 updated value: " << that->var2 << std::endl;
+        return (that->var2 * that->var1);
         }
-        std::cout << "U's <#name2#> updated value: " << that-><#name2#> << std::endl;
-        return that-><#name2#> * that-><#name1#>;
+        return *inputValue; 
```

Comment:
This indentation is messed up. 
Also, you're returning `*inputValue` but it might be nullptr.  
it's probably not smart to dereference a `nullptr`!

Reply:
agreed and understood, but not sure what to return.....